examples/all-clusters-app/esp32/main/include/DeviceWithDisplay.h

```diff
@@ -0,0 +1,94 @@
+/*
+ *
+ *    Copyright (c) 2022 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include "BluetoothWidget.h"
+#include "Button.h"
+#include "CHIPDeviceManager.h"
+#include "DeviceCallbacks.h"
+#include "Display.h"
+#include "Globals.h"
+#include "LEDWidget.h"
+#include "ListScreen.h"
+#include "OpenThreadLaunch.h"
+#include "QRCodeScreen.h"
+#include "ScreenManager.h"
+#include "ShellCommands.h"
+#include "StatusScreen.h"
+#include "WiFiWidget.h"
+#include "esp_heap_caps_init.h"
+#include "esp_log.h"
+#include "esp_netif.h"
+#include "esp_spi_flash.h"
+#include "esp_system.h"
+#include "esp_wifi.h"
+#include "freertos/FreeRTOS.h"
+#include "freertos/task.h"
+#include "nvs_flash.h"
+#include "platform/PlatformManager.h"
+#include "shell_extension/launch.h"
+
+#include <cmath>
+#include <cstdio>
+#include <ctype.h>
+#include <string>
+#include <vector>
+
+#include <app-common/zap-generated/att-storage.h>
+#include <app-common/zap-generated/attribute-id.h>
+#include <app-common/zap-generated/attribute-type.h>
+#include <app-common/zap-generated/attributes/Accessors.h>
+#include <app-common/zap-generated/cluster-id.h>
+#include <app/clusters/door-lock-server/door-lock-server.h>
+#include <app/clusters/network-commissioning/network-commissioning.h>
+#include <app/clusters/on-off-server/on-off-server.h>
+#include <app/clusters/ota-requestor/BDXDownloader.h>
+#include <app/clusters/ota-requestor/OTARequestor.h>
+#include <app/server/AppDelegate.h>
+#include <app/server/Dnssd.h>
+#include <app/server/OnboardingCodesUtil.h>
+#include <app/server/Server.h>
+#include <app/util/af-types.h>
+#include <app/util/af.h>
+#include <binding-handler.h>
+#include <credentials/DeviceAttestationCredsProvider.h>
+#include <credentials/examples/DeviceAttestationCredsExample.h>
+#include <lib/shell/Engine.h>
+#include <lib/support/CHIPMem.h>
+#include <lib/support/ErrorStr.h>
+#include <platform/CHIPDeviceLayer.h>
+#include <platform/ESP32/NetworkCommissioningDriver.h>
+#include <platform/ESP32/OTAImageProcessorImpl.h>
+#include <platform/GenericOTARequestorDriver.h>
+#include <setup_payload/ManualSetupPayloadGenerator.h>
+#include <setup_payload/QRCodeSetupPayloadGenerator.h>
```

Comment:
That's really a lot of headers for the following macros and fucntion declaration. Having lots of includes in a common header will cause slow compile, please include required headers in each source file.